Foundation/ISO8601DateFormatter.swift

```diff
@@ -79,8 +79,44 @@ open class ISO8601DateFormatter : Formatter, NSSecureCoding {
         super.init()
     }
     
-    public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
-    open override func encode(with aCoder: NSCoder) { NSUnimplemented() }
+    public required init?(coder aDecoder: NSCoder) {
+        guard aDecoder.allowsKeyedCoding else {
+            fatalError("ISO8601DateFormatter cannot be decoded by non-keyed archivers")
+        }
+        
+        let options = Options(rawValue: UInt(aDecoder.decodeInteger(forKey: "NS.formatOptions")))
+        let timeZone: NSTimeZone?
+        
+        if aDecoder.containsValue(forKey: "NS.timeZone") {
+            if let tz = aDecoder.decodeObject(of: NSTimeZone.self, forKey: "NS.timeZone") {
+                timeZone = tz
+            } else {
+                aDecoder.failWithError(CocoaError(.coderReadCorrupt, userInfo: [ NSLocalizedDescriptionKey: "Time zone was corrupt while decoding ISO8601DateFormatter" ]))
+                return nil
+            }
+        } else {
+            timeZone = nil
+        }
+        
+        self.formatOptions = options
+        if let zone = timeZone?._swiftObject {
+            self.timeZone = zone
+        }
+        
+        super.init()
+    }
+    
+    open override func encode(with aCoder: NSCoder) {
+        guard aCoder.allowsKeyedCoding else {
+            fatalError("Encoder does not allow key encoding")
```

Comment:
The wording of this suggests that the formatter _can't_ be formatted by key encoders. It may be clearer to borrow the language from `init()`'s `guard`, such as `"ISO8601DateFormatter can be encoded only by keyed archivers"`.

Reply:
I'll rewrite this. (It matches the Darwin exception message, but we have freedom here.)